### PR TITLE
updated NEWS

### PR DESCRIPTION
**Pull request by mdehoon**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Adding a NEWS entry describing the changes in Bio.Seq.

### CODE REVIEW

File: NEWS.rst [hunk 1]
```diff
@@ -14,6 +14,99 @@ The latest news is at the top of this file.
...
+object, which will be converted to a ``bytes`` or ``bytearray`` object assuming
+an ASCII encoding. Alternatively, a ``bytes`` or ``bytearray`` object can be
+used, or an instance of any class inheriting from the new
+``SequenceDataAbstractBaseClass`` abstract base class in ``Bio.Seq``. This
+requires that the class implements the ``__len__`` and ``__getitem`` methods
+that return the sequence length and sequence contents on demand. Initialzing a
+``Seq`` instance using an instance of a class inheriting from
+``SequenceDataAbstractBaseClass`` allows the ``Seq`` object to be lazy, meaning
+that its sequence is provided on demand only, without requiring to initialize
+the full sequence. This feature is now used in `BioSQL`, providing on-demand
```
Comment (peterjc) [inline NEWS.rst:70]: Double backticks on BioSQL?

File: NEWS.rst [hunk 2]
```diff
@@ -14,6 +14,99 @@ The latest news is at the top of this file.
...
+``Seq`` objects a string object was used, and a Unicode array object for
+``MutableSeq`` objects. This was maintained during the transition from Python2
+to Python3. However, a Python2 string object corresponds to a ``bytes`` object
+in Python3, storing the string as a series of 256-bit characters. While non-
+ASCII characters could be stored in Python2 strings, they were not treated as
+such. For example:
+
+In Python2:
+
+.. code-block:: python
```
Comment (peterjc) [inline NEWS.rst:28]: You need to indent the code sample if using a "code block", thus:

NEWS.rst:28: D000 Content block expected for the "code-block" directive; none found.

Same for the other examples.

File: NEWS.rst [hunk 3]
```diff
@@ -14,6 +14,99 @@ The latest news is at the top of this file.
...
+that its sequence is provided on demand only, without requiring to initialize
+the full sequence. This feature is now used in `BioSQL`, providing on-demand
+sequence loading from an SQL database, as well as in a new parser for twoBit
+(.2bit) sequence data added to ``Bio.SeqIO``. This is a lazy parser that allows
+fast access to genome-size DNA sequence files by not having to read the full
+genome sequence. The new ``_UndefinedSequenceData`` class in ``Bio.Seq``  also
+inherits from ``SequenceDataAbstractBaseClass`` to represent sequences of known
+length but unknown sequence contents. This provides an alternative to
+``UnknownSeq``, which is now deprecated as its definition was ambiguous. For
+example, in these examples the ``UnknownSeq`` is interpreted as a sequence with
```
Comment (peterjc) [inline NEWS.rst:78]: From the pre-commit linting with doc8,

NEWS.rst:78: D002 Trailing whitespace

Comment (peterjc) [reply NEWS.rst:78]: That extra space also likely triggers this:

NEWS.rst:78: D001 Line too long

File: NEWS.rst [hunk 4]
```diff
@@ -14,6 +14,99 @@ The latest news is at the top of this file.
...
+<class 'str'>
+>>> len(s)
+8
+
+In Python3, storing the sequence contents as ``bytes`` and ``bytearray`` objects
+has the further advantage that both support the buffer protocol.
+
+Taking advantage of the similarity between ``bytes`` and ``bytearray``, the
+``Seq`` and ``MutableSeq`` classes now inherit from an abstract base class
+``_SeqAbstractBaseClass`` in ``Bio.Seq`` that implements most of the ``Seq`` and
```
Comment (peterjc) [inline NEWS.rst:51]: doc8 grumbles:

NEWS.rst:51: D001 Line too long

File: NEWS.rst [hunk 5]
```diff
@@ -14,6 +14,99 @@ The latest news is at the top of this file.
...
+
+.. code-block:: python
+
+>>> s = "Генетика"
+>>> type(s)
+<class 'str'>
+>>> len(s)
+8
+
+In Python3, storing the sequence contents as ``bytes`` and ``bytearray`` objects
```
Comment (peterjc) [inline NEWS.rst:46]: doc8 grumbles:

NEWS.rst:46: D001 Line too long

**Comment by peterjc** [comment]
This is an unusually long NEWS entry, and very technical. It could certainly make the basis of a good blog post, but I wonder if the NEWS file is the best place?

Perhaps the Tutorial chapters ``Doc/Tutorial/chapter_seq_objects.tex`` and ``Doc/Tutorial/chapter_advanced.tex`` for the technical bits?

**Comment by mdehoon** [comment]
> This is an unusually long NEWS entry, and very technical. It could certainly make the basis of a good blog post, but I wonder if the NEWS file is the best place?
> 
> Perhaps the Tutorial chapters `Doc/Tutorial/chapter_seq_objects.tex` and `Doc/Tutorial/chapter_advanced.tex` for the technical bits?

Sure, feel free to move it if you think it fits in better elsewhere.

**Pull request merged**